packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts

```diff
@@ -211,13 +215,40 @@ export async function returnQuoteFromAltMMAsync<ResponseT>(
         }
     }
 
-    const response = await axiosInstance.post(`${url}/quotes`, data, {
-        headers: { Authorization: `Bearer ${apiKey}` },
-        timeout: maxResponseTimeMs,
-    });
+    // let response: AxiosResponse<any>;
+    const response = await axiosInstance
+        .post(`${url}/quotes`, data, {
+            headers: { Authorization: `Bearer ${apiKey}` },
+            timeout: maxResponseTimeMs,
+        })
+        .catch(err => {
+            warningLogger(err, `Alt RFQ MM request failed`);
+            throw new Error(`Alt RFQ MM request failed`);
+        });
+
+    // empty response will get filtered out in validation
+    const emptyResponse = {};
 
+    // tslint:disable-next-line:custom-no-magic-numbers
+    if (response.status !== SUCCESS_CODE) {
+        const rejectedRequestInfo = {
+            status: response.status,
+            message: response.data,
+        };
+        warningLogger(rejectedRequestInfo, `Alt RFQ MM did not return a status of ${SUCCESS_CODE}`);
+        return {
+            data: (emptyResponse as unknown) as ResponseT,
+            status: response.status,
+        };
+    }
+    // successful handling but no quote is indicated by status = 'rejected'
     if (response.data.status === 'rejected') {
-        throw new Error('alt MM rejected quote');
+        return {
+            data: (emptyResponse as unknown) as ResponseT,
+            // hack: set the http status to 204 no content so we can more
+            // easily track when no quote is returned
+            status: 204,
```

Comment:
A small thing - I do think the hack will be easy to trip on in the future. If possible, could we add a log statement aboe this, similar to what we have for the if case above?

Reply:
good idea! added a warning log